src/workspace-data/upload-data/UploadData.js

```diff
@@ -762,10 +766,54 @@ const MetadataUploadPanel = ({
     }
   });
 
+  const handleAutoGeneratedFile = (autoGeneratedFile, eventName, eventStatus) => {
+    const isDone = eventStatus !== 'generated';
+    if (autoGeneratedFile) {
+      void Metrics().captureEvent(eventName, {
+        ...autoGeneratedFile,
+        status: eventStatus,
+      });
+      isDone && setAutoGeneratedFile(null);
+    }
+  };
+
+  const doAutoGenerate = _.flow(Utils.withBusyState(setAutoGenerating))(async () => {
+    try {
+      const workspace = Workspaces().workspace(namespace, name);
+
+      setAutoGeneratedFile({
+        workspaceNamespace: namespace,
+        workspaceName: name,
+        entityType: collection,
```

Comment:
would it be better to ask the user to specify the table name - perhaps with a suggested default of the `collection` value - instead of assuming? I fear that the subdirectory into which users upload files is not what they desire for a table name.

Reply:
I think it's ok to start with this for preview and something we can get feedback on!